job_progress/job_progress.py

```diff
@@ -120,17 +120,62 @@ def __exit__(self, exc_type, exc_value, traceback):
         if self.delete_on_closing:
             self.delete()
 
-    def add_one_progress_state(self, state):
+    def add_one_progress_state(self, state, item_id=None):
         """Add one unit status."""
-        return self.backend.add_one_progress_state(self.id, state)
+        return self.backend.add_one_progress_state(self.id, state, item_id)
 
-    def add_one_failure(self):
-        """Add one failure state."""
-        return self.add_one_progress_state(states.FAILURE)
+    def add_one_failure(self, item_id=None):
+        """Add one failure state.
 
-    def add_one_success(self):
-        """Add one success state."""
-        return self.add_one_progress_state(states.SUCCESS)
+        :param item_id: Details with this progress. If it is None, then no
```

Comment:
The comment here is not super clear. Unclear what details means in this context. Perhaps add an example?